source/develop/release-management/features/storage/pass-discard-from-guest-to-underlying-storage.html.md

```diff
@@ -0,0 +1,235 @@
+
+* The actual size of *disk1* reduces.
+* The amount of free space in *sd1* gets bigger.
+* *vm2* can write to *disk2*.
+
+### Restrictions
+
+#### The underlying storage
+
+##### Block storage
+The underlying storage must support *discard* calls in order for this to work.<br/>
+That's why when trying to enable *"Pass Discard"* for a VM disk, the engine first queries the disk's storage domain for its *discard* support.
+
+This is how we know if the storage supports *discard*:
+
+- Let *lunX* be a device and *dm-X* be its corresponding dm device for a natural number X. Then *lunX* is considered to ***support discard*** iff the value of `/sys/block/dm-X/queue/discard_max_bytes` is bigger than 0 [^1].
+- A storage domain *supports discard* iff **all** of the luns that it is built from *support discard*. That means that if at least one of its luns doesn't *support discard*, then so is the storage domain.
+
+##### File storage
+Since the shared file system adds another layer between the guest's file system and the underlying shared block device, the engine can't check if the block device *supports discard*. Therefore, *Pass Discard* can be enabled for any disk on any file storage domain, and it will actually work only if the underlying file system and block device support it.<br/>
+Thus, *Pass Discard* on file storage works on the basis of **best effort**.
+
+#### The virtual machine's disk
+It is possible to enable *"Pass Discard"* for two disk storage types: **Images** and **Direct LUNs**.<br/>
+Regardless the type, the disk interfaces that support *"Pass Discard"* are **VirtIO SCSI** and **IDE**.
+
+Note that for *Direct LUNs* with a *VirtIO SCSI* interface, enabling the *"Enable SCSI Pass-Through"* property already gives the disk *discard* support (among other things).<br/>
+We still need to decide about the relationship between these two fields - it should be discussed with the UX team (**<font color="red"> TBD </font>**).
+
+#### The guest virtual machine
+> **Note**<br/>
+> The following restrictions cannot be satisfied by the engine (at least not at this point, see the [Future plans](#future-plans) section for more details), and should be handled by the VM's owner.
+
+In order to send a *discard* command from the guest, one can do it:
+
+- Manually - using the `fstrim` linux command.
+- Automatically on file removal (*online discarding*) - by adding the `discard` option to the list of the device's mount options. For example, in EL based guests' `/etc/fstab`:
+
+```properties
+/dev/mapper/rhel-root   /   ext4    defaults,discard    1   1
+# Here's the discard option -----------------^
+```
+
+For more information about sending *discard* from EL7 based OSs, refer to [Storage Administration Guide - Discard unused blocks](https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/7/html/Storage_Administration_Guide/ch02s04.html).<br/>
```

Comment:
Unfortunately I haven't found one.
It seems like Fedora stopped to release a "Storage Administration Guide" since Fedora 15.